movai_core_shared/core/message_client.py

```diff
@@ -82,20 +70,38 @@ def send_request(
                 "robot_info": self._robot_info,
             }
         }
+        return request
 
-        self._zmq_client.send(request)
-        if respose_required:
-            raw_response = self._zmq_client.recieve()
+    def _extract_response(self, msg):
```

Comment:
this function name does not reflect really what it does, maybe it should be prepare_response maybe

Reply:
why is does exactly what it meanse it extract thr reponse from the message

Reply:
on the contrary, it adds response key in case does not exist